openrtb_ext/bidders.go

```diff
@@ -564,10 +564,42 @@ type BidderParamValidator interface {
 	Schema(name BidderName) string
 }
 
+type paramsValidatorHelper interface {
+	readDir(name string) ([]os.DirEntry, error)
+	readFile(name string) ([]byte, error)
+	newReferenceLoader(source string) gojsonschema.JSONLoader
+	newSchema(l gojsonschema.JSONLoader) (*gojsonschema.Schema, error)
+	abs(path string) (string, error)
+}
+
+type paramsHelper struct{}
+
+func (paramsHelper) readDir(name string) ([]os.DirEntry, error) {
+	return os.ReadDir(name)
+}
+
+func (paramsHelper) readFile(name string) ([]byte, error) {
+	return os.ReadFile(name)
+}
+
+func (paramsHelper) newReferenceLoader(source string) gojsonschema.JSONLoader {
+	return gojsonschema.NewReferenceLoader(source)
+}
+
+func (paramsHelper) newSchema(l gojsonschema.JSONLoader) (*gojsonschema.Schema, error) {
+	return gojsonschema.NewSchema(l)
+}
+
+func (paramsHelper) abs(path string) (string, error) {
+	return filepath.Abs(path)
+}
+
+var paramsValidator paramsValidatorHelper = paramsHelper{}
+
 // NewBidderParamsValidator makes a BidderParamValidator, assuming all the necessary files exist in the filesystem.
 // This will error if, for example, a Bidder gets added but no JSON schema is written for them.
 func NewBidderParamsValidator(schemaDirectory string) (BidderParamValidator, error) {
-	fileInfos, err := os.ReadDir(schemaDirectory)
+	fileInfos, err := paramsValidator.readDir(schemaDirectory)
```

Comment:
Instead of declaring `paramsValidator` as package/global var. Update `NewBidderParamsValidator` to take `paramsValidatorHelper` interface as input. Caller of  `NewBidderParamsValidator` can then pass `paramsValidator` or `mockParamsHelper` as implementor

Reply:
Well, in that case we would need to make `paramsValidatorHelper` interface and `paramsHelper` struct public since caller of the `NewBidderParamsValidator` function is outside this package. IMO, caller should not worry about the implementation details of `NewBidderParamsValidator` function and should be handled within the package itself.